### PR TITLE
Refactor the switch delegate APIs from platform layer to application domain

### LINKED ISSUE #14624: Events being sent on all the endpoints
#### Problem
 * expected behavior: Events triggered on the device should be sent only to a particular endpoint_id.
 * actual behavior: Events triggered are sent to all the endpoints that have the cluster included.
 * steps to reproduce: Create a device with 2 endpoints and both having the switch cluster. Trigger the SwitchLatched event on the device. The event gets reported on both the endpoints instead of just one.
 * reference: Check `OnSwitchLatched` in `switch-server.cpp`.

 #### Proposed Solution
 * The events should accept the endpoint_id also as an argument, and send the event just on that endpoint.

### PR DESCRIPTION
**Pull request by yufengwangca**
#### Problem
What is being fixed?  Examples:
* expected behavior: Events triggered on the device should be sent only to a particular endpoint_id.
actual behavior: Events triggered are sent to all the endpoints that have the cluster included.
* Fixes #14624

#### Change overview
Refactor the switch delegate APIs from platform layer to application domain and event is only triggered on a particular endpoint_id.

TODO: Implement switch event trigger on one of existing example app

#### Testing
How was this tested? (at least one bullet point required)
* Tested with tested app with switch cluster enabled

### CODE REVIEW

File: src/app/clusters/switch-server/switch-server.h [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace Switch {
+
+/**
+ * @brief switch-server class
+ */
+class SwitchServer
```
Comment (bzbarsky-apple) [inline src/app/clusters/switch-server/switch-server.h:34]: `Switch::SwitchServer` is a bit odd.  Why not just `Switch::Server`?

File: src/app/clusters/switch-server/switch-server.h [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+     * @brief
+     *   Called when the latching switch is moved to a new position.
+     */
+    void OnSwitchLatched(EndpointId endpoint, uint8_t newPosition);
+
+    /**
+     * @brief
+     *   Called when the momentary switch starts to be pressed.
+     */
+    void OnInitialPressed(EndpointId endpoint, uint8_t newPosition);
```
Comment (bzbarsky-apple) [inline src/app/clusters/switch-server/switch-server.h:49]:
```suggestion
    void OnInitialPress(EndpointId endpoint, uint8_t newPosition);
```
"Initial" is not being "Pressed"...

File: src/app/clusters/switch-server/switch-server.h [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+    /**
+     * @brief
+     *   Called when the momentary switch has been pressed for a "long" time.
+     */
+    void OnLongPressed(EndpointId endpoint, uint8_t newPosition);
```
Comment (bzbarsky-apple) [inline src/app/clusters/switch-server/switch-server.h:55]:
```suggestion
    void OnLongPress(EndpointId endpoint, uint8_t newPosition);
```

File: src/app/clusters/switch-server/switch-server.h [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+    /**
+     * @brief
+     *   Called when the momentary switch has been released.
+     */
+    void OnShortReleased(EndpointId endpoint, uint8_t previousPosition);
```
Comment (bzbarsky-apple) [inline src/app/clusters/switch-server/switch-server.h:61]:
```suggestion
    void OnShortRelease(EndpointId endpoint, uint8_t previousPosition);
```

File: src/app/clusters/switch-server/switch-server.h [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+    /**
+     * @brief
+     *   Called when the momentary switch has been released (after debouncing)
+     *   and after having been pressed for a long time.
```
Comment (bzbarsky-apple) [inline src/app/clusters/switch-server/switch-server.h:66]:
```suggestion
     *   after having been pressed for a long time.
```

File: src/app/clusters/switch-server/switch-server.h [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+     */
+    void OnLongReleased(EndpointId endpoint, uint8_t previousPosition);
```
Comment (bzbarsky-apple) [inline src/app/clusters/switch-server/switch-server.h:68]:
```suggestion
    void OnLongRelease(EndpointId endpoint, uint8_t previousPosition);
```

**Review by bzbarsky-apple — APPROVED** [review]

**Review by yunhanw-google — APPROVED** [review]

**Pull request merged**